index.js

```diff
@@ -72,14 +73,16 @@ const resolver = Consumer.create({
         requestType = 'GET';
         const getCaseResponse = await getCase.fetch();
         const isCaseFound = (getCaseResponse.exists ? 'found' : 'not found');
-        logger.info({ externalID, message: 'Casework GET request successful. externalId:' +
-          `${externalID} was ${isCaseFound}`});
+        logger.info({
+          externalID, message: 'Casework GET request successful. externalId:' +
+            `${externalID} was ${isCaseFound}`
+        });
         caseID = getCaseResponse.caseId;
 
         if (!getCaseResponse.exists) {
           requestType = 'CREATECASE';
           const data = await submitCase.save();
-          caseID = data.createcaseresponse.caseid;
+          caseID = data.data.createcaseresponse.caseid;
```

Comment:
you can destructure it I think
```suggestion
const { caseid: caseID } = data?.data?.createcaseresponse || {};

if (!caseID) {
  logger.warn({ message: 'Failed to extract Case ID', data });
}

```
You can also use Nullish Coalescing and Optional Chaining to deal with errors in the inside the object.

you can also log an error if the caseID can't be extracted

Reply:
Added this is now